### ISSUE TITLE
concatenate in 'vertical' dimension

### ISSUE THREAD

**Issue by alevillard** [vlfeat/matconvnet#510]
Hi, I want to use the layer dagnn.Concat but not to concatenate orizontally the outputs of others layer, but vertically, one above another. Is it possible? I see in file vl_nnconcat(inputs, dim) that I could maybe choose the direction of concatenation, as cat matlab command have. But I'm not sure if it's correct or how to write in the layer

add.layer('name', dagnn.Concat(), {'in1' ' 'in2'}, {output}; 
%%this is for 'orizontal' concatenation, but I would like a vertical one. how can I  do this? thank you

**Comment by lenck** [comment]
Yeah, that should work. I'm not sure what you mean by 'vertically' as by default the concatenation is along the 3rd dimension (channels) so it is not either horizontally or vertically. But basically concatenating along the rows is dim 1, along columns (side by side) is dim 2. And you can simply do it by adding a layer as:
```
add.layer('name', dagnn.Concat('dim', <DIM>), {'in1' ' 'in2'}, {output}; 
```

**Comment by XuYunqiu** [comment]
hi，
I got 2 stream data: 30*20*8 and 30*20*8
how can I use dagnn.Concat() to get the output 30*40*8
